storage/index/convert/convert.go

```diff
@@ -77,6 +77,24 @@ func FromMetric(id ident.ID, tags ident.Tags) (doc.Document, error) {
 	}, nil
 }
 
+// FromMetricNoClone converts the provided metric id+tags into a document without cloning.
+func FromMetricNoClone(id ident.ID, tags ident.Tags) (doc.Document, error) {
```

Comment:
there's similar code in the shard too. would be good to consolidate into a single place. How about `storage/index/convert` ? 

/cc @robskillington

Reply:
lol doh. nvm i see the filename

Reply:
Maybe make the shard code re-use this too

Reply:
@prateek The shard logic has the special tag subslice stuff...should we move that logic into the helpers so its not only on series insertion? Not sure about doing that in this P.R

Reply:
Yeah it'd be good have all code-paths re-use the logic. Don't think there's anything specific to shard/series in that code. Feel free to do in a subsequent PR if that's easier.